Libraries/LibGUI/GSlider.cpp

```diff
@@ -56,15 +56,19 @@ void GSlider::paint_event(GPaintEvent& event)
 Rect GSlider::knob_rect() const
 {
     auto inner_rect = this->inner_rect();
-    float range_size = m_max - m_min;
-    float adjusted_value = m_value - m_min;
-    float relative_value = adjusted_value / range_size;
-    Rect rect {
-        inner_rect.x() + (int)(relative_value * inner_rect.width()) - knob_width() / 2,
-        0,
-        knob_width(),
-        knob_height()
-    };
+    Rect rect;
+    rect.set_y(0);
+    rect.set_height(knob_height());
+
+    if (m_max - m_min) {
+        float scale = inner_rect.width() / (m_max - m_min + 1);
```

Comment:
We need to cast the dividend and divisor to `float` as well, or we will lose precision:

```suggestion
        float scale = (float)inner_rect.width() / (float)(m_max - m_min + 1);
```

You can see how this breaks otherwise if you run the `wg` application and try to drag the topmost slider to the far right.